test/e2e/nodepool_nto_machineconfig_test.go

```diff
@@ -46,222 +50,335 @@ spec:
 
-	nodePools := &hyperv1.NodePoolList{}
-	if err := client.List(ctx, nodePools, crclient.InNamespace(hostedCluster.Namespace)); err != nil {
-		t.Fatalf("failed to list nodepools in namespace %s: %v", hostedCluster.Namespace, err)
-	}
+		e2eutil.WaitForImageRollout(t, ctx, mgmtClient, hostedCluster, globalOpts.LatestReleaseImage)
 
-	var nodePool hyperv1.NodePool
-	for _, nodePool = range nodePools.Items {
-		if nodePool.Spec.ClusterName != hostedCluster.Name {
-			continue
+		tuningConfigConfigMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "hugepages-tuned-test",
+				Namespace: hostedCluster.Namespace,
+			},
+			Data: map[string]string{tuningConfigKey: hugepagesTuned},
+		}
+		if err := mgmtClient.Create(ctx, tuningConfigConfigMap); err != nil {
+			if !errors.IsAlreadyExists(err) {
+				t.Fatalf("failed to create configmap for custom Tuned object: %v", err)
+			}
 		}
 
 		np := nodePool.DeepCopy()
 		nodePool.Spec.TuningConfig = append(nodePool.Spec.TuningConfig, corev1.LocalObjectReference{Name: tuningConfigConfigMap.Name})
-		if err := client.Patch(ctx, &nodePool, crclient.MergeFrom(np)); err != nil {
+		if err := mgmtClient.Patch(ctx, nodePool, crclient.MergeFrom(np)); err != nil {
 			t.Fatalf("failed to update nodepool %s after adding Tuned config: %v", nodePool.Name, err)
 		}
-	}
 
-	ds := ntoMachineConfigUpdatedVerificationDS.DeepCopy()
-	if err := guestClient.Create(ctx, ds); err != nil {
-		t.Fatalf("failed to create %s DaemonSet in guestcluster: %v", ds.Name, err)
-	}
+		// DS Customization
+		ds := ntoMachineConfigUpdatedVerificationDS.DeepCopy()
+		dsName := ds.Name + "-replace"
 
-	t.Logf("waiting for rollout of NodePools with NTO-generated config")
-	err = wait.PollImmediateWithContext(ctx, 5*time.Second, 15*time.Minute, func(ctx context.Context) (bool, error) {
-		if ctx.Err() != nil {
-			return false, err
-		}
-		pods := &corev1.PodList{}
-		if err := guestClient.List(ctx, pods, crclient.InNamespace(ds.Namespace), crclient.MatchingLabels(ds.Spec.Selector.MatchLabels)); err != nil {
-			t.Logf("WARNING: failed to list pods, will retry: %v", err)
-			return false, nil
-		}
-		nodes := &corev1.NodeList{}
-		if err := guestClient.List(ctx, nodes); err != nil {
-			t.Logf("WARNING: failed to list nodes, will retry: %v", err)
-			return false, nil
-		}
-		if len(pods.Items) != len(nodes.Items) {
-			return false, nil
+		for _, c := range ds.Spec.Template.Spec.Containers {
+			if c.Name == ds.Name {
+				c.Name = dsName
+			}
 		}
 
-		for _, pod := range pods.Items {
-			if !isPodReady(&pod) {
-				return false, nil
+		ds.Name = dsName
+		ds.ObjectMeta.Labels = make(map[string]string)
+		ds.ObjectMeta.Labels["hypershift.openshift.io/nodePool"] = nodePool.Name
+
+		ds.Spec.Selector.MatchLabels["name"] = dsName
+		ds.Spec.Template.ObjectMeta.Labels["name"] = dsName
+		ds.Spec.Template.ObjectMeta.Labels["hypershift.openshift.io/nodePool"] = nodePool.Name
```

Comment:
why don't use hypershift.openshift.io/nodePool in MatchLabels as well?

Reply:
Makes totally sense